File: goldbach.py
def is_prime(x):
    i=2
    while(i<=x/2):
        if x%i==0 :
            return False
        i=i+1
    return True

n=int(input("enter number"))
i=2
j=n-2
while(j>=2):
    if is_prime(j)==True and is_prime(i)==True:
        print(str(n) +"->"+str(i)+"+"+str(j))
        break
    j=j-1
    i=i+1

File: Goldbach.java
import java.util.Scanner;
class Goldbach {
    public static boolean isPrime(int x) {

        for (int i = 2; i <= x / 2; i++) {
            if (x % i == 0) {
                return false;
            }
        }
        return true;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.println("enter number");
        int n=scanner.nextInt();
        int i=2;
        for (int j=n-2; j>=2; j--,i++) {
                if(isPrime(j)==true && isPrime(i)==true)
                {
                    System.out.println(n+"->"+i+"+"+j);
                    break;
                }
        }
        scanner.close();
    }
}